### PR TITLE
Game Logic, TypeScript, package-lock.json update

### PR DESCRIPTION
**Pull request by nmoadev**
* Added a `game-logic` module to the `src` folder. The goal is to capture the "rules" and state changes of the game so that components can observe the state, but only change it via a few functions that represent player actions.

* the `data` folder is how I imagine us defining the actual playing map and tiles.
* I used TypeScript because I find the interface definitions helpful

### CODE REVIEW

File: src/data/tile-catalog.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { TileCatalog } from '../game_logic';
+
+// This file implements a "database" of tiles, each with unique appearance, choices, etc
```
Comment (scottysseus) [inline src/data/tile-catalog.ts:3]: this will be REALLY fun to brainstorm :)

File: src/game_logic/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
+export interface Project {
+  projectName: string
+  description: string
+  tileName: string
+  cost: number
+  revenue: number
+  effort: number
+  happiness: number
+}
```
Comment (scottysseus) [inline src/game_logic/interfaces.ts:9]: I'd say go ahead and submit- we can keep iterating on this

I think we need to solidify the project process a little. I envision each project as transforming one tile into another - `Clear Debris` transforms `Overgrown Lot` into `Empty Lot`, `Landscape` turns `Empty Lot` into `Park`, and so on

`revenue` and `happiness` are properties of both tiles, not the project, and the player can see the change to those fields (and, crucially, the tags) when choosing which project to pursue.

Projects are simpler:

```TypeScript
export interface Project {
    name: string
    description: string
    toTile: string // tile ID?
    cost: number
    effort: number
}
```

**Review by scottysseus — APPROVED** [review]
looks good! thanks!

File: src/game_logic/interfaces.ts (hunk 2 above)
Comment (scottysseus) [reply src/game_logic/interfaces.ts:9]: let me know if this makes sense; I am also open to other interpretations of projects/contracts, but I would like to make sure I understand whatever one we go with

Comment (nmoadev) [reply src/game_logic/interfaces.ts:9]: this makes sense, and reduces the number of different entities that we have to understand. I hadn't thought of treating "unimproved" and "improved" tiles as just types of tile. Good idea. Also makes it easier for content creators to just generate a bunch of ideas

**Pull request merged**